lib/matplotlib/axes/_axes.py

```diff
@@ -4879,22 +4816,11 @@ def fill_betweenx(self, y, x1, x2=0, where=None,
                 raise ValueError('Input passed into argument "%r"' % name +
                                  'is not 1-dimensional.')
 
-        if x1.ndim == 0:
-            x1 = np.ones_like(y) * x1
-        if x2.ndim == 0:
-            x2 = np.ones_like(y) * x2
-
         if where is None:
-            where = np.ones(len(y), np.bool)
-        else:
-            where = np.asarray(where, np.bool)
-
-        if not (y.shape == x1.shape == x2.shape == where.shape):
-            raise ValueError("Argument dimensions are incompatible")
+            where = True
+        where = where & ~reduce(np.logical_or, map(np.ma.getmask, [y, x1, x2]))
```

Comment:
Thinking more about `reduce` "going away" in Python 3 more than legibility, but I guess we can just do `from functools import reduce` instead of `six.moves`.